src/kata4.humanCatDogYears.js

```diff
@@ -1,3 +1,25 @@
-const humanCatDogYears = (number) => {};
+const humanCatDogYears = (number) => {
+  const humanYears = number;
+  function catYears(humanAge) {
+    if (humanAge <= 1) {
+      return 15;
+    }
+    if (humanAge <= 2) {
+      return 24;
+    }
+    return 24 + (humanAge - 2) * 4;
+  }
+
+  function dogYears(humanAge) {
+    if (humanAge <= 1) {
+      return 15;
+    }
+    if (humanAge <= 2) {
+      return 24;
+    }
+    return 24 + (humanAge - 2) * 5;
+  }
+  return [humanYears, catYears(number), dogYears(number)];
+};
```

Comment:
So theres a few things to look at here. First off, the logic here will fail if you pass in 0 to the function as it will return `[0,15,15]` when it should return `[0,0,0]`. You can fix this by putting something called an early exit condition at the start of the function:
```
  if (number === 0) return [0, 0, 0];
```
Next, theres no point in assigning the value of `number` to a new `humanYears` variable, you can just use `number` as it is

We should also use the same function syntax throughout, since the main function itself is written in arrow syntax the internal functions should be too. Also, you should check the value against a specific number in the if statements if it only applies to one value, eg:
```
  const catYears = (humanAge) => {
    if (humanAge === 1) {
      return 15;
    }
    if (humanAge === 2) {
      return 24;
    }
    return 24 + (humanAge - 2) * 4;
  };
```
Putting all this together, you end up with this:
```
const humanCatDogYears = (number) => {
  if (number === 0) return [0, 0, 0];

  const catYears = (humanAge) => {
    if (humanAge === 1) {
      return 15;
    }
    if (humanAge === 2) {
      return 24;
    }
    return 24 + (humanAge - 2) * 4;
  };

  const dogYears = (humanAge) => {
    if (humanAge === 1) {
      return 15;
    }
    if (humanAge === 2) {
      return 24;
    }
    return 24 + (humanAge - 2) * 5;
  };
  return [number, catYears(number), dogYears(number)];
};
```

Just a bit tidier :)

Reply:
Makes sense, thank you!